### PR TITLE
WIP: Add testing resources for React

### PR DESCRIPTION
**Pull request by danielburnley**

### CODE REVIEW

File: core-skills/web-application-development-with-react/testing/components.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+### Example one - A component which displays a name
+
+Testing a component which takes a single prop, name, and displays it on the page.
+
+**NameDisplay/index.js**
+
+```jsx
+import React from "react";
+
+export default props => <div data-test="name">{props.name}</div>;
```
[the remark this answers is not in the document]
Comment (danielburnley) [reply core-skills/web-application-development-with-react/testing/components.md]: React tends to advise functional components unless you need state or the like IIRC - though yeah maybe worth a section on pure function components vs class components

File: core-skills/web-application-development-with-react/testing/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Testing in React Web Applications
+
+<< internal note - this paragraph could be expressed better >>
+
+React web applications will require slightly different test strategies to testing APIs.
+Although the testing of the vanilla Javascript classes (e.g. Usecase/Gateways) will be
+similar to how you'd test them in backend systems, testing React components themselves
+requires strategies you'd apply to more typical frontend testing.
```
[the remark this answers is not in the document]
Comment (danielburnley) [reply core-skills/web-application-development-with-react/testing/README.md]: Yeah that's my plan further down in the test examples, then I plan to use the same examples in another section around Single Responsibility Components

File: core-skills/web-application-development-with-react/testing/components.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+### What not to test
+
+Frontend tests can be brittle if you test the wrong things, for example, what a header says. If this is
+static text that isn't tied to any behaviour you shouldn't test this, as it could be changed and cause
+test failures.
+
+### Enzyme - Mount vs Shallow
+
+<< mount vs shallow, mount is more consistent as shallow is similar to mocking classes you don't own >>
```
Comment (ccp92) [inline core-skills/web-application-development-with-react/testing/components.md]: No sure I agree with this. Where possible, we should be testing against shallow as mount will render all of the child components which means you're already not testing a component in isolation.
If there's something in the child component that needs to be tested, it should be done in the child component's tests.

File: core-skills/web-application-development-with-react/testing/components.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ccp92) [reply core-skills/web-application-development-with-react/testing/components.md]: Following changes being made to this skill, engineers working on the react core skill should be able to identify and use functional components

**Pull request closed without merging**